n2y/plugins/jinjarenderpage.py

```diff
@@ -190,21 +220,32 @@ def __init__(self, client, notion_data, page, get_children=True):
                 url_property=export_defaults["url_property"],
             )
 
-    def _get_database_ids_from_mentions(self):
-        for rich_text in self.caption:
-            is_mention = isinstance(rich_text, MentionRichText)
-            if is_mention and isinstance(rich_text.mention, DatabaseMention):
-                yield rich_text.mention.notion_database_id
-
-    def to_pandoc(self):
-        jinja_environment = self.client.plugin_data['jinjarenderpage']['environment']
+    def _render_text(self):
+        jinja_environment = self.client.plugin_data[
+            'jinjarenderpage'][self.page.notion_id]['environment']
         jinja_code = self.rich_text.to_plain_text()
         context = {
             "databases": self.databases,
             "page": self.page.properties_to_values(self.pandoc_format),
         }
+        if 'render_content' in jinja_code:
+            def render_content(notion_id):
+                page = self.client.get_page(notion_id)
+                for child in page.block.children:
+                    if isinstance(child, HeadingBlock):
+                        child.level += 1
```

Comment:
Instead of hard-coding the header indent at `+= 1` here, I think we should add an optional second argument to the `render_content` filter that defaults to 0, but can be set to any positive or negative number. If a negative number is used, we need to be careful to ensure the resulting header level is larger than 1. Thus, if the header level is 3 and a header adjustment of -3 is passed in, then the result should still be 1. Perhaps:

```
child.level = max(1, child.level + level_adjustment)
```

Reply:
I think it would make more sense to default it to 1 so that the default functionality is to maintain consistency with the page headers as a whole but otherwise I agree.

Reply:
I think the default should be to keep the content as-is, since I wouldn't expect a filter like this to automatically indent and it could surprise people if it mutates the content automatically. If we kept the +1, then we'd probably want to change the name of the filter to indicate this behavior, but then the name may be too long. (That said, having this option is a great idea.)